### DISCUSSION TITLE
Pixelart not scaling as expected

### DISCUSSION THREAD

**Discussion by TaylorAnderson** [HaxeFlixel/flixel#2336] [Q&A]
It's been a long while so I've forgotten the terminology for this, but....I have a game with a zoom level of 4.  Everything works fine, until I start rotating sprites and...they're not pixelated the way I expected them to be. 
![image](https://user-images.githubusercontent.com/5308535/123723950-7b76bc00-d859-11eb-8cae-7153836237d9.png)
 
How do I fix this? I've tried disabling antialiasing on the camera and the sprite itself, and setting StageQuality to LOW, but thats not fixing it.

**Comment by RichardBray** [comment]
I've noticed you've asked a similar question in the Discord server which is good. They tend to answer questions there faster than the forums. 

Have you looked into @AustinEast's pixel-perfect rendering technique?

https://gist.github.com/AustinEast/108ec6601cf47389fdee81e0014cdc2a
https://github.com/AustinEast/haxeflixel-pixel-perfect

Not something I've tried personally but if it works for him I'm sure it could work for you. Don't know if it'll be okay for HTML5 though 🤷‍♂️

**Reply by jobf** [reply]
I hit this problem last year and couldn't find a way to disable the smooth scaling in OpenFL for HTML5 - I think that was the problem anyway. 

I ended up implementing a shader that enlarges the image and keeps it pixelated in HTML5. It might be considered a hack but it works for me. Check my repo here (there is a demo you can see there too) - https://github.com/jobf/haxeflixel-pixel-scaling-shader

Alternatively I can vouch for Austin East's solution - works great if you're compiling for Windows for example.

edit: Someone on discord just mentioned another approach to this - https://github.com/aeveis/haxeflixel-pixel-parity-shader

**Reply by TaylorAnderson** [reply]
Hey, I did try the solution mentioned on the Discord, and it did (partially) work, but introduced some other visual weirdness.  Just gonna pause on the pixelation front for now haha, it's not a huge deal.  For non-rotated stuff its basically perfect as-is, and I don't expect to do a lot of rotation.
